libmscore/rendermidi.cpp

```diff
@@ -899,6 +1024,61 @@ static QList<NoteEventList> renderChord(Chord* chord, int gateTime, int ontime)
       return ell;
       }
 
+    
+void Score::createGraceNotesPlayEvents(QList<Chord*> gnb, int tick, Chord* chord, int &ontime){
+    int n = gnb.size();
+    if (n) {
+        //
+        //  render grace notes:
+        //  simplified implementation:
+        //  - grace notes start on the beat of the main note
+        //  - duration: appoggiatura: 0.5  * duration of main note (2/3 for dotted notes, 4/7 for double-dotted)
+        //              acciacatura: min of 0.5 * duration or 65ms fixed (independent of duration or tempo)
+        //  - for appoggiaturas, the duration is divided by the number of grace notes
+        //  - the grace note duration as notated does not matter
+        //
+        Chord* graceChord = gnb[0];
+        if (graceChord->noteType() ==  NoteType::ACCIACCATURA) {
+            qreal ticksPerSecond = tempo(tick) * MScore::division;
+            int graceTimeMS = 65 * n;     // value determined empirically (TODO: make instrument-specific, like articulations)
+            // 1000 occurs below for two different reasons:
+            // number of milliseconds per second, also unit for ontime
+            qreal chordTimeMS = (chord->actualTicks() / ticksPerSecond) * 1000;
+            ontime = qMin(500, static_cast<int>((graceTimeMS / chordTimeMS) * 1000));
+        }
+        else if (chord->dots() == 1) {
```

Comment:
It's not an issue but a forum post - https://musescore.org/en/node/52716

Justification is clear enough to me: the commonly stated advice of 2/3 length for appoggiaturas on dotted notes probably presupposes you are in compound meter.  If not, then maybe that rule shouldn't apply.  On the other hand, maybe it should.  Other than some YouTube recording of one particular piece that do indeed support the idea, I don't see anything solid to go on here.